x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsRelocationIntegTests.java

```diff
@@ -114,7 +115,19 @@ public void testRelocationWaitsForPreWarm() throws Exception {
         assertBusy(() -> assertThat(getActiveRelocations(restoredIndex), Matchers.empty()));
     }
 
+    private static List<RecoveryState> getRelocations(String restoredIndex) {
+        return getRelocationsStream(restoredIndex).collect(Collectors.toList());
+    }
+
     private static List<RecoveryState> getActiveRelocations(String restoredIndex) {
+        return getRelocationsStream(restoredIndex)
+            // filter for relocations that are not in stage FINALIZE (they could end up in this stage without progress for good if the
+            // target node does not have enough cache space available to hold the primary completely
+            .filter(recoveryState -> recoveryState.getStage() != RecoveryState.Stage.FINALIZE)
```

Comment:
I am not sure I follow this, how do we run into not having enough space available in this test?

Reply:
The cache size is set randomly in (https://github.com/elastic/elasticsearch/blob/36683a4581a8fc2f108701d92af2e4d527e56d02/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/BaseSearchableSnapshotsIntegTestCase.java#L53) and in some cases it's too small to hold all the data.